builtin/providers/openstack/types.go

```diff
@@ -154,15 +155,42 @@ func (lrt *LogRoundTripper) formatJSON(raw []byte) string {
 	return string(pretty)
 }
 
+// Firewall is an OpenStack firewall.
+type Firewall struct {
+	firewalls.Firewall
+	routerinsertion.FirewallExt
+}
+
 // FirewallCreateOpts represents the attributes used when creating a new firewall.
 type FirewallCreateOpts struct {
-	firewalls.CreateOpts
+	firewalls.CreateOptsBuilder
```

Comment:
:+1: 

I recently ran into this with a different type in another PR.

In addition to this, I wonder if this field should be explicitly labeled `CreateOptsBuilder` so [this type of format](https://github.com/hashicorp/terraform/blob/master/builtin/providers/openstack/resource_openstack_compute_instance_v2.go#L415-L420) can work? I haven't tried to modify the custom types here to support the field name. Just a thought.